R/drive_publish.R

```diff
@@ -19,41 +19,42 @@ drive_publish <- function(file = NULL,
                           verbose = TRUE) {
   file_update <- drive_check_publish(file = file, verbose = FALSE)
 
-  request <- build_drive_publish(file = file_update, publish = publish, ...)
+  x <- list(...)
+  x$published <- publish
+  if (!("publishAuto" %in% names(x))) {
+    x$publishAuto <- TRUE
+  }
+
+  x$fileId <- file_update$id
+
+  x$revisionId <- file_update$publish$revision
+
+  if (grepl("application/vnd.google-apps.spreadsheet", file_update$kitchen_sink$mimeType)) {
+    x$revisionId <- 1
+  }
+
+  request <- build_request(
+    endpoint = "drive.revisions.update",
+    params = x
+  )
   response <- make_request(request, encode = "json")
   proc_res <- process_drive_publish(response = response,
                                     file = file_update,
                                     verbose = verbose)
 
-  ##for some reason we have to make the request 2x?
+  ## if we want to autopublish, it must have already been published, so
+  ## we need to run again
+  if (x$published == TRUE & x$publishAuto == TRUE) {
```

Comment:
If `x$published` and `x$publishAuto` are logical, which I assume they are, this should just be `x$published && x$publishAuto`. Note the use of `&&`. If you are worried about NAs, then you could do `isTRUE(x$published) && isTRUE(x$publishAuto)`.